bundles/org.openhab.binding.tibber/src/main/java/org/openhab/binding/tibber/internal/handler/TibberHandler.java

```diff
@@ -0,0 +1,471 @@
+
+                    String timestampDailyFrom = (myObject.get("from").toString()).substring(1, 20);
+                    updateState(DAILY_FROM, new DateTimeType(timestampDailyFrom));
+
+                    String timestampDailyTo = (myObject.get("to").toString()).substring(1, 20);
+                    updateState(DAILY_TO, new DateTimeType(timestampDailyTo));
+
+                    updateChannel(DAILY_COST, myObject.get("cost").toString());
+                    updateChannel(DAILY_CONSUMPTION, myObject.get("consumption").toString());
+
+                } catch (JsonSyntaxException e) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                            "Error communicating with Tibber API: " + e.getMessage());
+                }
+            }
+            if (jsonResponse.contains("hourly")) {
+                try {
+                    JsonObject myObject = (JsonObject) object.getAsJsonObject("data").getAsJsonObject("viewer")
+                            .getAsJsonObject("home").getAsJsonObject("hourly").getAsJsonArray("nodes").get(0);
+
+                    String timestampHourlyFrom = (myObject.get("from").toString()).substring(1, 20);
+                    updateState(HOURLY_FROM, new DateTimeType(timestampHourlyFrom));
+
+                    String timestampHourlyTo = (myObject.get("to").toString()).substring(1, 20);
+                    updateState(HOURLY_TO, new DateTimeType(timestampHourlyTo));
+
+                    updateChannel(HOURLY_COST, myObject.get("cost").toString());
+                    updateChannel(HOURLY_CONSUMPTION, myObject.get("consumption").toString());
+
+                } catch (JsonSyntaxException e) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                            "Error communicating with Tibber API: " + e.getMessage());
+                }
+            }
+
+        } else if (jsonResponse.contains("error")) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "Error in response from Tibber API: " + jsonResponse);
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Unexpected response from Tibber");
+        }
+    }
+
+    public void startRefresh(int refresh) {
+        if (pollingJob == null) {
+            pollingJob = scheduler.scheduleWithFixedDelay(() -> {
+                try {
+                    updateRequest();
+                } catch (IOException e) {
+                    logger.warn("IO Exception: ", e);
+                }
+            }, 1, refresh, TimeUnit.MINUTES);
+        }
+    }
+
+    public void updateRequest() throws IOException {
+        getURLInput(BASE_URL);
+
+        if ("true".equals(rtEnabled) && !isConnected()) {
+            try {
+                open();
+            } catch (Exception e) {
```

Comment:
Can you catch specific exceptions instead?

Reply:
Corrected.